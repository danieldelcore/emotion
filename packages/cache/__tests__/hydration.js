// @flow
import { safeQuerySelector } from 'test-utils'
import hashString from '@emotion/hash'
import createCache from '@emotion/cache'

beforeEach(() => {
  safeQuerySelector('head').innerHTML = ''
  safeQuerySelector('body').innerHTML = ''
})

test('it works', () => {
  let css = `color:hotpink;`
  let hash = hashString(css)
  safeQuerySelector(
    'body'
  ).innerHTML = `<style data-emotion="css ${hash}">.css-${hash}{${css}}</style>`
  let cache = createCache({ key: 'css' })
  expect(cache.inserted).toEqual({ [hash]: true })
  expect(document.documentElement).toMatchSnapshot()
})

test('rehydrated styles to head can be flushed', () => {
  let css = `color:hotpink;`
  let hash = hashString(css)
  safeQuerySelector(
    'head'
  ).innerHTML = `<style data-emotion="emo ${hash}">.emo-${hash}{${css}}</style>`

  // this moves emotion style tags at initialization time
  jest.resetModules()
  require('@emotion/react')

  let cache = createCache({ key: 'emo' })
  expect(document.documentElement).toMatchSnapshot()
  cache.sheet.flush()
  expect(document.documentElement).toMatchSnapshot()
})

test('Should only mount style elements matching the cache key', () => {
  let css = `color:hotpink;`
  let hash = hashString(css)

  safeQuerySelector(
    'body'
  ).innerHTML = `<style data-emotion="emo ${hash}">.css-${hash}{${css}}</style>`

  const cache = createCache({ key: 'css' })

  expect(document.documentElement).toMatchSnapshot()
  expect(cache.inserted).toEqual({})

  const cache2 = createCache({ key: 'emo' })
  expect(cache2.inserted).toEqual({ [hash]: true })
  expect(document.documentElement).toMatchSnapshot()
})
