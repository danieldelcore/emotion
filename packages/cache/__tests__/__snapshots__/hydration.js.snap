// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should only mount style elements matching the cache key 1`] = `
<html>
  <head />
  <body>
    <style
      data-emotion="emo 1lrxbo5"
    >
      .css-1lrxbo5{color:hotpink;}
    </style>
  </body>
</html>
`;

exports[`Should only mount style elements matching the cache key 2`] = `
<html>
  <head>
    <style
      data-emotion="emo 1lrxbo5"
    >
      .css-1lrxbo5{color:hotpink;}
    </style>
  </head>
  <body />
</html>
`;

exports[`it works 1`] = `
<html>
  <head>
    <style
      data-emotion="css 1lrxbo5"
      data-s=""
    >
      .css-1lrxbo5{color:hotpink;}
    </style>
  </head>
  <body />
</html>
`;

exports[`rehydrated styles to head can be flushed 1`] = `
<html>
  <head>
    <style
      data-emotion="emo 1lrxbo5"
    >
      .emo-1lrxbo5{color:hotpink;}
    </style>
  </head>
  <body />
</html>
`;

exports[`rehydrated styles to head can be flushed 2`] = `
<html>
  <head />
  <body />
</html>
`;
